packages/main/src/plugin/proxy-system.ts

```diff
@@ -0,0 +1,141 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { promisify } from 'node:util';
+
+import { type ProxySettings } from '@podman-desktop/api';
+import WinReg from 'winreg';
+
+import { isLinux, isMac, isWindows } from '../util.js';
+import type { Proxy } from './proxy.js';
+import { Exec } from './util/exec.js';
+
+export async function getProxySettingsFromSystem(proxy: Proxy): Promise<ProxySettings> {
+  if (isWindows()) {
+    return getWindowsProxySettings();
+  } else if (isMac()) {
+    return getMacOSProxySettings(new Exec(proxy));
+  } else if (isLinux()) {
+    const httpProxy = process.env.HTTP_PROXY;
+    const httpsProxy = process.env.HTTPS_PROXY;
+    const noProxy = process.env.NO_PROXY;
+    return { httpProxy, httpsProxy, noProxy };
+  }
+  return {} as ProxySettings;
+}
+
+async function getWindowsProxySettings(): Promise<ProxySettings> {
+  try {
+    let enabled = false;
+    let httpProxy = undefined;
+    let httpsProxy = undefined;
+    let noProxy = undefined;
+    const internetSettingsKey = new WinReg({
+      hive: WinReg.HKCU,
+      key: '\\Software\\Microsoft\\Windows\\CurrentVersion\\Internet Settings',
+    });
+    const keys = await promisify(internetSettingsKey.values).bind(internetSettingsKey)();
+    keys.forEach(key => {
+      if (key.name === 'ProxyEnable') {
+        enabled = key.value === '0x1';
+      } else if (key.name === 'ProxyServer') {
+        const items = key.value.split(';');
+        items.forEach(item => {
+          if (item.startsWith('http=')) {
+            httpProxy = `http://${item.substring(5)}`;
+          } else if (item.startsWith('https=')) {
+            httpsProxy = `http://${item.substring(6)}`;
+          } else {
+            httpProxy = httpsProxy = `http:${item}`;
+          }
+        });
+      } else if (key.name === 'ProxyOverride') {
+        noProxy = key.value;
+      }
+    });
+    return enabled ? { httpProxy, httpsProxy, noProxy } : ({} as ProxySettings);
```

Comment:
no {}

Reply:
Yes this is for the case the system information tells proxy is disabled so an empty structure is returned (just like manual and not data specified)